components/rhs_root_post/rhs_root_post.jsx

```diff
@@ -235,6 +239,27 @@ export default class RhsRootPost extends React.PureComponent {
         const isSystemMessage = PostUtils.isSystemMessage(post);
         const isMeMessage = ReduxPostUtils.isMeMessage(post);
 
+        const showRecentlyUsedReactions = (!isReadOnly && !isEphemeral && !post.failed && !isSystemMessage && !channelIsArchived && this.props.oneClickReactionsEnabled);
+        let showRecentReacions;
+        if (showRecentlyUsedReactions) {
+            const state = store.getState();
+            const recentEmojis = getRecentEmojis(state).slice(-3);
+            const emojiMap = getEmojiMap(state);
+            const emojis = [];
+            for (let i = 0; i < recentEmojis.length; i++) {
+                emojis.push(emojiMap.get(recentEmojis[i]));
+            }
```

Comment:
we should only consume redux store in a connect function for class components.
This is usually located in the component's `index` file.

You could also write a selector to re-use:

e.g:
```ts
const getNamingIsHardEmojis = (state) => {
    const recentEmojis = getRecentEmojis(state).slice(-3);
    const emojiMap = getEmojiMap(state);
    const emojis = [];
    for (let i = 0; i < recentEmojis.length; i++) {
        emojis.push(emojiMap.get(recentEmojis[i]));
    }
    
    return emojis;
}
```